sa/sa_test.go

```diff
@@ -3472,8 +3472,8 @@ func TestLeaseSpecificCRLShard(t *testing.T) {
 	)
 	test.AssertError(t, err, "leasing unavailable shard")
 
```

Comment:
Here's a patch to test the conditional at the top of the ssa `leaseSpecificCRLShard` method and that a row was inserted and could be selected for a previously unknown shard.
```
index 7e9f79cee..26ce090bc 100644
--- a/sa/sa_test.go
+++ b/sa/sa_test.go
@@ -3414,6 +3414,18 @@ func TestLeaseSpecificCRLShard(t *testing.T) {
                LeasedUntil time.Time `db:"leasedUntil"`
        }
 
+       // Leasing more than one shard should fail.
+       _, err = sa.leaseSpecificCRLShard(
+               context.Background(),
+               &sapb.LeaseCRLShardRequest{
+                       IssuerNameID: 1,
+                       MinShardIdx:  1,
+                       MaxShardIdx:  2,
+                       Until:        timestamppb.New(until),
+               },
+       )
+       test.AssertError(t, err, "did not lease one specific shard")
+
        // Leasing an unleased shard should work.
        res, err := sa.leaseSpecificCRLShard(
                context.Background(),
@@ -3474,7 +3486,7 @@ func TestLeaseSpecificCRLShard(t *testing.T) {
 
        // Leasing a previously-unknown specific shard should work (to ease the
        // transition into using leasing).
-       _, err = sa.leaseSpecificCRLShard(
+       res, err = sa.leaseSpecificCRLShard(
                context.Background(),
                &sapb.LeaseCRLShardRequest{
                        IssuerNameID: 1,
@@ -3484,6 +3496,13 @@ func TestLeaseSpecificCRLShard(t *testing.T) {
                },
        )
        test.AssertNotError(t, err, "leasing unknown shard")
+       err = sa.dbMap.SelectOne(
+               &untilModel,
+               `SELECT leasedUntil FROM crlShards WHERE issuerID = ? AND idx = ? LIMIT 1`,
+               res.IssuerNameID,
+               res.ShardIdx,
+       )
+       test.AssertNotError(t, err, "could not find newly inserted shard")
```